// Copyright 2019 Zeppelin Bend Pty Ltd
// This file is part of cimbend.
// 
// cimbend is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// cimbend is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
// 
// You should have received a copy of the GNU Affero General Public License
// along with cimbend.  If not, see <https://www.gnu.org/licenses/>.


syntax = "proto3";
option java_multiple_files = true;
option java_package = "com.zepben.protobuf.cim.iec61970.base.wires";
package zepben.cim.iec61970.base.wires;

/**
 * Vector group of the transformer for protective relaying, e.g., Dyn1. For unbalanced transformers, this may not be
 * simply determined from the constituent winding connections and phase angle dispacements.
 * <p>
 * The vectorGroup string consists of the following components in the order listed: high voltage winding connection,
 * mid voltage winding connection (for three winding transformers), phase displacement clock number from 0 to 11, low
 * voltage winding connection phase displacement clock number from 0 to 11.
 * <p>
 * The winding connections are D (delta), Y (wye), YN (wye with neutral), Z (zigzag), ZN (zigzag with neutral),
 * A (auto transformer). Upper case means the high voltage, lower case mid or low. The high voltage winding always has
 * clock postion 0 and is not included in the vector group string.
 * <p>
 * Some examples:
 *                YNy0 (two winding wye to wye with no phase displacement),
 *                YNd11 (two winding wye to delta with 330 degrees phase displacement),
 *                YNyn0d5 (three winding transformer wye with neutral high voltgage, wye with neutral mid
 *                         voltgage and no phase displacement, delta low voltage with 150 degrees displacement).
 * <p>
 * Phase displacement is defined as the angular difference between the phasors representing the voltages between the
 * neutral point (real or imaginary) and the corresponding terminals of two windings, a positive sequence voltage system
 * being applied to the high-voltage terminals, following each other in alphabetical sequence if they are lettered, or
 * in numerical sequence if they are numbered: the phasors are assumed to rotate in a counter-clockwise sense.
 */
enum VectorGroup {
    UNKNOWN = 0;
    DD0 = 1;
    DZ0 = 2;
    DZN0 = 3;
    YNY0 = 4;
    YNYN0 = 5;
    YY0 = 6;
    YYN0 = 7;
    ZD0 = 8;
    ZND0 = 9;
    DYN1 = 10;
    DZ1 = 11;
    DZN1 = 12;
    YD1 = 13;
    YND1 = 14;
    YNZN1 = 15;
    YZ1 = 16;
    YZN1 = 17;
    ZD1 = 18;
    ZND1 = 19;
    ZNYN1 = 20;
    ZY1 = 21;
    ZYN1 = 22;
    DY5 = 23;
    DYN5 = 24;
    YD5 = 25;
    YND5 = 26;
    YNZ5 = 27;
    YNZN5 = 28;
    YZ5 = 29;
    YZN5 = 30;
    ZNY5 = 31;
    ZNYN5 = 32;
    ZY5 = 33;
    ZYN5 = 34;
    DD6 = 35;
    DZ6 = 36;
    DZN6 = 37;
    YNY6 = 38;
    YNYN6 = 39;
    YY6 = 40;
    YYN6 = 41;
    ZD6 = 42;
    ZND6 = 43;
    DY7 = 44;
    DYN7 = 45;
    DZ7 = 46;
    DZN7 = 47;
    YD7 = 48;
    YND7 = 49;
    YNZN7 = 50;
    YZ7 = 51;
    YZN7 = 52;
    ZD7 = 53;
    ZND7 = 54;
    ZNYN7 = 55;
    ZY7 = 56;
    ZYN7 = 57;
    DY11 = 58;
    DYN11 = 59;
    YD11 = 60;
    YND11 = 61;
    YNZ11 = 62;
    YNZN11 = 63;
    YZ11 = 64;
    YZN11 = 65;
    ZNY11 = 66;
    ZNYN11 = 67;
    ZY11 = 68;
    ZYN11 = 69;
    DY1 = 70;
    Y0 = 71;
    YN0 = 72;
    D0 = 73;
    ZNY1 = 74;
    ZNY7 = 75;
    DDN0 = 76;
    DND0 = 77;
    DNYN1 = 78;
    DNYN11 = 79;
    YNDN1 = 80;
    YNDN11 = 81;
}
